src/engine/errors.ts

```diff
@@ -30,6 +31,17 @@ export class LimitsExceededError extends ContentfulError {
   }
 }
 
+export class LimitsExceededForApplyError extends ContentfulError {
+  constructor(context: ApplyChangesetContext) {
+    const message = `The detected number of entries to be changed is too high.\nThe currently allowed limit is ${context.limit} entries.`
+    const details = {
+      limit: context.limit,
```

Comment:
Currently there is no limit set. 
It should be set [here](https://github.com/contentful/contentful-merge/blob/06e5e2832f0e7c5a81f6fd1d86a9662033d1a3bd/src/commands/apply/index.ts#L66) (and not coming from a flag).